creator/ingest_runs/data_generator/study_generator.py

```diff
@@ -0,0 +1,504 @@
+        )
+        rows = []
+        for gi in range(self.total_specimens):
+            # Harmonized
+            ext = ".g.vcf.gz"
+            h = True
+            fp = make_file(gi, ext, harmonized=h)
+            row_dict = {
+                "sample_id": f"SM-{gi}",
+                "data_type": "WGS",
+                "filepath": s3_path(os.path.split(fp)[-1], harmonized=h),
+            }
+            # Unharmonized
+            for ei, extension in enumerate(self.source_gf_extensions):
+                h = False
+                fp = make_file(f'{gi}-{ei}', extension, harmonized=h)
+                row_dict = row_dict.copy()
+                # Use fixed ids for source gfs
+                row_dict["kf_id_genomic_file"] = generate_kf_id()
+                row_dict["project_id"] = f"SE-{row_dict['sample_id']}"
+                row_dict["source_read"] = s3_path(
+                    os.path.split(fp)[-1], harmonized=h
+                )
+                rows.append(row_dict)
+
+        df = pd.DataFrame(rows)
+        df.to_csv(manifest_fp, sep='\t', index=False)
+        self.logger.info(f"Wrote {manifest_fp}")
+
+        return df
+
+    def _upload_gfs_to_s3(self):
+        """
+        Upload gfs to S3 study bucket
+        """
+        self.logger.info(
+            f"Uploading genomic files in {self.data_dir} to S3 bucket "
+            f"{self.study_bucket}"
+        )
+
+        s3_client = boto3.client('s3')
+        gf_manifest = read_df(os.path.join(self.data_dir, "gf_manifest.tsv"))
+
+        s3_paths = sorted(
+            list(gf_manifest["filepath"].drop_duplicates().values) +
+            list(gf_manifest["source_read"].drop_duplicates().values)
+        )
+        for s3_path in s3_paths:
+            suffix = (
+                s3_path.split(f"s3://{self.study_bucket}")[-1].lstrip("/")
+            )
+            fp = os.path.join(self.data_dir, suffix)
+            try:
+                s3_client.upload_file(fp, self.study_bucket, suffix)
+            except ClientError as e:
+                self.logger.error(e)
+            self.logger.info(
+                f"Uploaded {suffix} to {s3_path}"
+            )
+
+    def _create_gf_s3_manifests(self, create=False):
+        """
+        Generate S3 object manifests for unharmonized and harmonized
+        genomic files
+        """
+        verb = "Creating" if create else "Reading"
+        self.logger.info(
+            f"{verb} S3 object manifest for genomic files in "
+            f"s3://{self.study_bucket}"
+        )
+        fp = os.path.join(self.data_dir, 's3_gf_manifest.tsv')
+        if not create:
+            return read_df(fp)
+
+        s3_data = fetch_aws_bucket_obj_info(
+            bucket_name=self.study_bucket,
+            search_prefixes=[
+                "source/genomic-files/",
+                "harmonized/genomic-files/"
+            ],
+        )
+        df = pd.DataFrame(s3_data)
+
+        # Filter out folders
```

Comment:
Had to do this since they removed that `drop_folders` kwarg from the fetch function